src/mono/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs

```diff
@@ -245,7 +245,7 @@ public static AssemblyBuilder DefineDynamicAssembly(AssemblyName name, AssemblyB
             return new AssemblyBuilder(name, access);
         }
 
-        [RequiresDynamicCode("Generating new code at runtime is not supported with native AOT.")]
+        [RequiresDynamicCode("Creating a DynamicMethod requires dynamic code.")]
```

Comment:
These two aren't for creating a DynamicMethod:

```suggestion
        [RequiresDynamicCode("Defining a dynamic assembly requires dynamic code.")]
```